singleton-service/src/service/proxy.rs

```diff
@@ -0,0 +1,155 @@
+    /// Message queue will get registered when on_vm_start callback gets called.
+    fn on_vm_start(&mut self, _vm_configuration_size: usize) -> bool {
+        if let Ok(q_id) = register_shared_queue(QUEUE_NAME) {
+            self.queue_id = Some(q_id);
+        }
+        // TODO : handle MQ failure, change info to trace after dev
+        info!(
+            "Registered new message queue with id: {}",
+            self.queue_id.unwrap()
+        );
+        true
+    }
+
+    /// Configuration passed by envoy.yaml will get deserialized to ServiceConfig. If there's an issue with the
+    /// passed configuration, default configuration will be used.
+    fn on_configure(&mut self, _config_size: usize) -> bool {
+        // Check for the configuration passed by envoy.yaml
+        let configuration: Vec<u8> = match self.get_configuration() {
+            Some(c) => c,
+            None => {
+                warn!("Configuration missing. Please check the envoy.yaml file for filter configuration");
+                return false;
+            }
+        };
+
+        // Parse and store the configuration passed by envoy.yaml
+        match serde_json::from_slice::<ServiceConfig>(configuration.as_ref()) {
+            Ok(config) => {
+                debug!("configuring {}: {:?}", self.context_id, config);
+                self.config = config;
+                true
+            }
+            Err(e) => {
+                warn!(
+                    "Failed to parse envoy.yaml configuration: {:?}. Using default configuration.",
+                    e
+                );
+                true
+            }
+        }
+    }
+
+    /// on_queue_ready will get triggered when cache filter enqueue data. dequeue_shared_queue() is used
+    /// to dequeu data from the queue. For each entry the following functions are performed.
+    ///     * If local cache update is required, update metrics and perform local cache update.
+    ///     * Add the entry to delta store.
+    fn on_queue_ready(&mut self, _queue_id: u32) {
+        match dequeue_shared_queue(self.queue_id.unwrap()) {
+            Ok(queue_entry) => {
+                // TODO : Handle local cache update based on the flag passed from cache filter (when it's implemented)
+                info!(
+                    "Consumed following message from the shared queue: {:?}",
+                    queue_entry
+                );
+                match queue_entry {
+                    Some(threescale) => {
+                        // TODO: Hanlde deserialization safely.
+                        let threescale: ThreescaleData = bincode::deserialize(&threescale).unwrap();
+                        // TODO: If update required, then update, Handle update failure
+                        self.delta_store.update_delta_store(&threescale);
+                    }
+                    None => {
+                        info!("No application found from the message queue entry")
```

Comment:
We are now consuming a Message struct(which contains the flag), not a Threescale struct. It's present inside `threescale::structs`.

Reply:
fixed